packages/platforms/quanta/x86-64/x86-64-quanta-ix1b-rglbmc/onlp/builds/src/x86_64_quanta_ix1b_rglbmc/module/src/thermali.c

```diff
@@ -0,0 +1,33 @@
+/************************************************************
+ * <bsn.cl fy=2014 v=onl>
+ *
+ *           Copyright 2014 Big Switch Networks, Inc.
+ *
+ * Licensed under the Eclipse Public License, Version 1.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *        http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *
+ * </bsn.cl>
+ ************************************************************
+ *
+ *
+ *
+ ***********************************************************/
+#include <onlp/platformi/thermali.h>
+#include "x86_64_quanta_ix1b_rglbmc_log.h"
+
+int
+onlp_thermali_init(void)
```

Comment:
@jonathantsai-qci I understand that a BMC is becoming more common on higher end systems but it is not always the case. It is not reinventing the wheel -- ONL must handle both BMC and non-BMC enabled systems and the ONLP interface must work the same on all platforms. Relying on ipmi tools as the primary interface is not possible. 

While its not possible to use IPMI directly I see no reason to reimplement this each time for each BMC-enabled system. I propose we implement a single BMC/IPMI ONLP implementation that can then be selected for platforms that use it. All reporting would then still be possible through the standard interfaces.